src/mlpack/methods/reinforcement_learning/replay/prioritized_replay.hpp

```diff
@@ -95,13 +117,33 @@ class PrioritizedReplay
    * @param reward Given reward.
    * @param nextState Given next state.
    * @param isEnd Whether next state is terminal state.
+   * @param discount The discount parameter.
    */
-  void Store(const StateType& state,
+  void Store(StateType state,
              ActionType action,
              double reward,
-             const StateType& nextState,
-             bool isEnd)
+             StateType nextState,
+             bool isEnd,
+             const double& discount)
   {
+    nStepBuffer.push_back({state, action, reward, nextState, isEnd});
+
+    // Single step transition is not ready.
+    if(nStepBuffer.size() < nSteps)
+      return;
+
+    // To keep the queue size fixed to nSteps.
+    if(nStepBuffer.size() > nSteps)
+      nStepBuffer.erase(nStepBuffer.begin());
```

Comment:
Might be better to use `std::deque` here, since it should be fast to add to the end and remove the front, let me know what you think.

Reply:
Yeah that should work. Thanks. Turns out I only need to change one word for that. :)

Reply:
Ohh, nice.